### ISSUE TITLE
Rendered Attributes Missing Space(s)

### ISSUE THREAD

**Issue by michaelsilver** [infolab-csail/WikipediaBase#41]
I have noticed that for some attribute mappings fetched with
```Python
>>> from wikipediabase.util import get_infobox, get_meta_infobox
>>> get_meta_infobox('<TEMPLATE_NAME>').rendered_keys()
```
the rendered attributes are missing spaces between words. For example:
```Python
>>> from wikipediabase.util import get_infobox,
get_meta_infobox
>>> get_meta_infobox('Template:Infobox weapon').rendered_keys()
{'origin': u'Place\xa0of\xa0origin',
'diameter': 'Diameter',
'traverse': 'Traverse',
'secondary_armament': 'Secondaryarmament',
'hilt_type': u'Hilt\xa0type',
'weight': 'Weight',
'max_range': u'Maximum\xa0firing\xa0range',
'guidance': 'Guidancesystem',
'armour': 'Armour',
'sights': 'Sights',
'number': u'Number\xa0built',
'height': 'Height',
'barrels': 'Barrels',
'rate': u'Rate\xa0of\xa0fire',
'caliber': 'Calibre',
'yield': u'Blast\xa0yield',
'range': u'Effective\xa0firing\xa0range',
'cartridge_weight': 'Shell weight',
'pw_ratio': 'Power/weight',
'speed': 'Speed',
'feed': u'Feed\xa0system',
'depth': 'Maximum depth',
'service': u'In\xa0service',
'production_date': 'Produced',
'altitude': 'Flight altitude',
'carriage': 'Carriage',
'crew': 'Crew',
'width': 'Width',
'primary_armament': 'Mainarmament',
'transport': 'Transport',
'suspension': 'Suspension',
'haft_type': u'Haft\xa0type',
'type': 'Type',
'detonation': 'Detonationmechanism',
'accuracy': 'Accuracy',
'engine': 'Engine',
'ceiling': 'Flight ceiling',
'designer': 'Designer',
'elevation': 'Elevation',
'propellant': 'Propellant',
'cartridge': 'Shell',
'blade_type': u'Blade\xa0type',
'used_by': u'Used\xa0by',
'wars': 'Wars',
'filling_weight': u'Warhead\xa0weight',
'boost': 'Boost time',
'recoil': 'Recoil',
'breech': 'Breech',
'sheath_type': 'Scabbard/sheath',
'variants': 'Variants',
'steering': 'Steeringsystem',
'manufacturer': 'Manufacturer',
'passengers': 'Passengers',
'launch_platform': 'Launchplatform',
'part_length': u'Barrel\xa0length',
'transmission': 'Transmission',
'head_type': u'Head\xa0type',
'fuel_capacity': u'Fuel\xa0capacity',
'design_date': 'Designed',
'wingspan': 'Wingspan',
'length': 'Length',
'vehicle_range': 'Operationalrange',
'filling': 'Warhead',
'engine_power': 'Engine',
'unit_cost': u'Unit\xa0cost',
'action': 'Action',
'velocity': u'Muzzle\xa0velocity',
'clearance': u'Ground\xa0clearance',
'payload_capacity': u'Payload\xa0capacity'}
```
These rendered attributes probably should have spaces but don't: `Secondaryarmament`, `Guidancesystem`, `Mainarmament`, `Detonationmechanism`, `Steeringsystem`, `Launchplatform`, `Operationalrange`. I have seen this problem in other infoboxes such as: `Template:Infobox firearm cartridge` (`Detonationmechanism`), `Template:Infobox military unit` (`Electronicwarfare`, `Notablecommanders`, `Currentcommander`), `Template:Infobox military conflict` (`Territorialchanges`). BTW, I'm not a war nut or anything... it just seems like this is happening with a lot of military related infoboxes :)

This is a particularly bad problem, because there is no way for me to just eliminate these attributes by regex matching.

This issue may be related to whether the rendered attribute gets split into two lines (with no space between when the linebreak is removed). See "Operational Range" in the infobox on this page: https://en.wikipedia.org/wiki/BM-21_Grad

I also notice that some rendered strings are unicode, and some are not -- don't know whether this is related to the problem, but worth noticing. FYI, I'm able to transform all the unicode string into clean non-unicode, properly turning `\xa0` into a space by using 
```Python
>>> from unidecode import unidecode
>>> unidecode('<unicode string>')
```

**Comment by michaelsilver** [comment]
Yup, looks like your merge fixed it. Thanks @fakedrake!

**Comment by fakedrake** [comment]
For the record the problem was that for lxml `toText` means literally remove all markup tags. `br` however modifies the text itself (adding newlines) so removing caused the problems you spotted. In other words the problem was that I did not anticipate that rendered attributes would contain hardcoded newlines.

**Comment by michaelsilver** [comment]
I am not sure whether this is related to commits to fix this issue or other commits following, but ever since this issue was closed, somehow the unicode changed so that what used to be `\xa0` is now `\xc2\xa0`. This is particularly problematic because unidecode changes `\xc2\xa0` to be `"A "` instead what used to happen which was `\xa0` decoding to `" "`:

```Python
>>> from wikipediabase.util import get_infobox, get_meta_infobox
>>> get_meta_infobox('Template:Infobox weapon').rendered_keys()
{'origin': u'Place\xc2\xa0of\xc2\xa0origin',
'diameter': 'Diameter',
'traverse': 'Traverse',
'secondary_armament': 'Secondary armament',
'hilt_type': u'Hilt\xc2\xa0type',
'weight': 'Weight',
'max_range': u'Maximum\xc2\xa0firing\xc2\xa0range',
'guidance': 'Guidance system',
'armour': 'Armour',
'sights': 'Sights',
'number': u'Number\xc2\xa0built',
'height': 'Height',
'barrels': 'Barrels',
'rate': u'Rate\xc2\xa0of\xc2\xa0fire',
'caliber': 'Calibre',
'yield': u'Blast\xc2\xa0yield',
'range': u'Effective\xc2\xa0firing\xc2\xa0range',
'cartridge_weight': 'Shell weight',
'pw_ratio': 'Power/weight',
'speed': 'Speed',
'feed': u'Feed\xc2\xa0system',
'depth': 'Maximum depth',
'service': u'In\xc2\xa0service',
'production_date': 'Produced',
'altitude': 'Flight altitude',
'carriage': 'Carriage',
'crew': 'Crew',
'width': 'Width',
'primary_armament': 'Main armament',
'transport': 'Transport',
'suspension': 'Suspension',
'haft_type': u'Haft\xc2\xa0type',
'type': 'Type',
'detonation': 'Detonation mechanism',
'accuracy': 'Accuracy',
'engine': 'Engine',
'ceiling': 'Flight ceiling',
'designer': 'Designer',
'elevation': 'Elevation',
'propellant': 'Propellant',
'cartridge': 'Shell',
'blade_type': u'Blade\xc2\xa0type',
'used_by': u'Used\xc2\xa0by',
'wars': 'Wars',
'filling_weight': u'Warhead\xc2\xa0weight',
'boost': 'Boost time',
'recoil': 'Recoil',
'breech': 'Breech',
'sheath_type': 'Scabbard/sheath',
'variants': 'Variants',
'steering': 'Steering system',
'manufacturer': 'Manufacturer',
'passengers': 'Passengers',
'launch_platform': 'Launch platform',
'part_length': u'Barrel\xc2\xa0length',
'transmission': 'Transmission',
'head_type': u'Head\xc2\xa0type',
'fuel_capacity': u'Fuel\xc2\xa0capacity',
'design_date': 'Designed',
'wingspan': 'Wingspan',
'length': 'Length',
'vehicle_range': 'Operational range',
'filling': 'Warhead',
'engine_power': 'Engine',
'unit_cost': u'Unit\xc2\xa0cost',
'action': 'Action',
'velocity': u'Muzzle\xc2\xa0velocity',
'clearance': u'Ground\xc2\xa0clearance',
'payload_capacity': u'Payload\xc2\xa0capacity'}
```

Any way to change the encoding back to `\xa0` instead of `\xc2\xa0`?

**Comment by michaelsilver** [comment]
Closed via #47
